### PR TITLE
[4.4] Vote : aggregateRating error in Google Rich Results

### PR DESCRIPTION
**Pull request by conseilgouz**
( see discussion https://github.com/joomla/joomla-cms/discussions/42920) and #42933

### Summary of Changes
When vote plugin is enabled, Rich results are created but are rejected by Google Rich Results tool.

### Testing Instructions
Enable Vote plugin.
Enable Voting in Articles component configuration.
Vote for one or more articles.
Check your article using https://search.google.com/test/rich-results

### Actual result BEFORE applying this Pull Request
Google Rich Results check gives an error 
Review snippets shows 2 errors : 
rich_snippet_error

### Expected result AFTER applying this Pull Request
Google Rich Results check is OK
rich_snippet_ok

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by conseilgouz** [comment]
@sandewt PR moved to 4.4 branch. 
Please test https://issues.joomla.org/tracker/joomla-cms/42934

File: plugins/content/vote/tmpl/rating.php [hunk 1]
```diff
@@ -79,6 +79,8 @@
 ?>
 <div class="content_rating" role="img" aria-label="<?php echo Text::sprintf('PLG_VOTE_STAR_RATING', $rating); ?>">
     <?php if ($rcount) : ?>
+    <div class="visually-hidden" itemscope="itemscope" itemtype="https://schema.org/Product">
+        <span itemprop="name"><?php echo $row->title;?></span>
```
Comment (sandewt) [inline plugins/content/vote/tmpl/rating.php]:
```suggestion
        <span itemprop="name"><?php echo $row->title; ?></span>
```

**Comment by ManuelVoileux** [comment]
I have tested this item :white_check_mark: successfully on 2f594e4b4492b4469bc36f9923f68eb188900f35<br /><br />I have experienced the same issue.
I have activated the Vote plugin, and I have used vote on an article.
When I check with the Google validation tool I get a critical error. itemReviewed field missing.
Google error test

AFTER applying this Pull Request I got green light from Google
OK Google index

As Fedik mentionned [Add the layout with Schema plugin](https://github.com/joomla/joomla-cms/pull/42933#issuecomment-1973429188 "Add the layout with Schema plugin")  so that no additionnal shema type (such as Product) is involved<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42934">issues.joomla.org/tracker/joomla-cms/42934</a>.</sub>

**Comment by ManuelVoileux** [comment]
I have experienced the same issue.
I have activated the Vote plugin, and I have used vote on an article.
When I check with the Google validation tool I get a critical error. itemReviewed field missing.
Google error test

AFTER applying this Pull Request I got green light from Google
OK Google index

As Fedik mentionned [Add the layout with Schema plugin](https://github.com/joomla/joomla-cms/pull/42933#issuecomment-1973429188 "Add the layout with Schema plugin")  so that no additionnal shema type (such as Product) is involved<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42934">issues.joomla.org/tracker/joomla-cms/42934</a>.</sub>

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 2f594e4b4492b4469bc36f9923f68eb188900f35<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42934">issues.joomla.org/tracker/joomla-cms/42934</a>.</sub>

**Comment by conseilgouz** [comment]
> As Fedik mentionned [Add the layout with Schema plugin](https://github.com/joomla/joomla-cms/pull/42933#issuecomment-1973429188) so that no additionnal shema type (such as Product) is involved

Schema plugins have been introduced in Joomla 5.0 and this PR is due to work with Joomla 4.4.

**Comment by ManuelVoileux** [comment]
You are right, sorry for the mistake. 
What is the appropriate process to have a J5 branch to solve this bug ?

**Comment by conseilgouz** [comment]
> You are right, sorry for the mistake. What is the appropriate process to have a J5 branch to solve this bug ?

It will be to move aggregatingRating into schema plugins, as each schema type has its own required fields list.

**Comment by sandewt** [comment]
for your information: Schema.org Product type plugin #42710

**Comment by conseilgouz** [comment]
> for your information: Schema.org Product type plugin #42710

Thank you for this information.

This PR is trying to fix an issue in vote plugin, it's not to create a schemaorg product plugin.

As stated in #42933 , I found we could make aggregateRating work by using product type.

In J5, I think we'll have to fix it another way, maybe through existing schemaorg plugins.

**Comment by sandewt** [comment]
Review versus Product ?

see https://github.com/joomla/joomla-cms/pull/42933#issuecomment-1975081570

**Comment by Fedik** [comment]
Sorry, the changes still incorrect. We should not force any `itemscope` in to the rating layout, 
It can be rendered as part of already defined `itemscope`. User can already have Article layout override with custom  `itemscope`. This changes will break his schema.

In the end, this hack replaces one problem to another.

For proper fix check the coments in PR for 5.1 (https://github.com/joomla/joomla-cms/pull/42933#issuecomment-1973429188)

**Comment by conseilgouz** [comment]
> Review versus Product ?
> 
> see [#42933 (comment)](https://github.com/joomla/joomla-cms/pull/42933#issuecomment-1975081570)

Review requires a single user, rating count will disappear.

**Comment by conseilgouz** [comment]
> For proper fix check the coments in PR for 5.1

I'm trying to implement 5.1 version of this hack using schemaorg plugins. 

I have a doubt on where to implement this.

For example, for Recipe, I added some lines into plugins/schema/recipe/src/Extension/Recipe.php so it added  Recipe dedicated rating lines to the Recipe information. I created a new class SchemaorgPrepareRecipeAggregateRating in libraries/src/Schemaorg.

Does it sound correct or is there a better place to add this ?

**Comment by sandewt** [comment]
> I'm trying to implement 5.1 version of this hack using schemaorg plugins.

Sorry, but I think it's better to focus on J4 here and not J5.

**Comment by Fedik** [comment]
> I'm trying to implement 5.1 version of this hack using schemaorg plugins.

In theory the voting plugin should listen to `onSchemaBeforeCompileHead` and apend the rating data.

**Comment by bembelimen** [comment]
> > I'm trying to implement 5.1 version of this hack using schemaorg plugins.
> 
> In theory the voting plugin should listen to `onSchemaBeforeCompileHead` and apend the rating data.
> 

Exactly that

**Comment by exlemor** [comment]
I have tested this item :white_check_mark: successfully on 07d5eed2499438e73b9fe406102d1b605711f632<br /><br />Hi @conseilgouz, I just tested this successfully just now.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42934">issues.joomla.org/tracker/joomla-cms/42934</a>.</sub>

**Comment by conseilgouz** [comment]
Hi Emmanuel,
Thank you for your test.
For this PR, there already were 2 human tests by @fgsw and @ManuelVoileux, but they disappeared from the counter.

**Comment by MacJoom** [comment]
Thank you!

**Comment by sandewt** [comment]
Sorry but this PR is not correct because the **Number of Votes** will never be visible.
Because the class is: class="visually-hidden".

Solution move the closing `</div`

**Pull request merged**